drivers/pwm/pwm_nrfx.c

```diff
@@ -10,88 +10,221 @@
 #include <logging/log.h>
 LOG_MODULE_REGISTER(pwm_nrfx);
 
-#define PWM_NRFX_CH_VALUE_NORMAL        (1UL << 15)
-#define PWM_NRFX_CH_VALUE_INVERTED      (0)
+#define PWM_NRFX_CH_POLARITY_POS        15      // Bit[15] is polarity, 1 means start period LOW
+#define PWM_NRFX_CH_COMPARE_MASK        0x7FFF  // Bit[14:0] is COMPARE
+#define PWM_PIN_NOT_CONNECTED           -1
 
 struct pwm_nrfx_config {
-	nrfx_pwm_t pwm;
-	nrfx_pwm_config_t config;
-	nrf_pwm_sequence_t seq;
+	NRF_PWM_Type * pwm_regs;                            // PWM Instance
```

Comment:
I don't entirely understand the need of completely rewriting the driver and no longer using nrfx as the underlying layer. Could you elaborate on this a bit?

Reply:
As discussed; will be reverting to using the nrfx layer as reasons for rewrite are no longer implemented.